src/parser/TranspileState.ts

```diff
@@ -54,12 +54,12 @@ export class TranspileState {
     /**
      * Shorthand for creating a new source node
      */
-    public sourceNode(locatable: { range?: Range }, code: string | SourceNode | Array<string | SourceNode>): SourceNode | undefined {
+    public sourceNode(locatable: { range?: Range }, code: string | SourceNode | Array<string | SourceNode>): SourceNode {
         return new SourceNode(
             //convert 0-based range line to 1-based SourceNode line
-            locatable.range.start.line + 1,
+            locatable.range ? locatable.range.start.line + 1 : null,
             //range and SourceNode character are both 0-based, so no conversion necessary
-            locatable.range.start.character,
+            locatable.range ? locatable.range.start.character : null,
```

Comment:
`SourceNode` is allowed to be passed `null` here according to the VSCode typedef file, so I chose to do that rather than risk an error being thrown if the range is undefined.